plugins/action_remove_user_org/src/removeUserOrg.ts

```diff
@@ -1,34 +1,58 @@
 import { ActionQueueStatus } from '../../../src/generated/graphql'
 import { ActionPluginInput } from '../../types'
+import grantPermissionsDBMethods from '../../action_grant_permissions/src/databaseMethods'
+import { UserOrg } from '../../../src/types'
 
 const removeUserOrg = async function ({ parameters, DBConnect }: ActionPluginInput) {
-  const { userId, orgId, user_id, organisation_id, org_id, deletePermissions = true } = parameters
-  const userOrg = {
-    userId: userId ?? user_id,
-    orgId: orgId ?? org_id ?? organisation_id,
-  }
+  const { username, orgName, deletePermissions = true } = parameters
+  const dbCommon = grantPermissionsDBMethods(DBConnect)
+
   try {
-    console.log(`\Removing user from organisation...`)
-    const result = await DBConnect.removeUserOrg(userOrg)
+    const userId =
+      parameters?.userId === null
+        ? null
+        : parameters?.userId ??
+          parameters?.user_id ??
+          (await dbCommon.getUserIdFromUsername(username))
+
+    const orgId =
+      parameters?.orgId === null
+        ? null
+        : parameters?.orgId ??
+          parameters?.org_id ??
+          parameters?.organisation_id ??
+          (await dbCommon.getOrgIdFromOrgname(orgName))
+
+    if (!orgId)
+      return {
+        status: ActionQueueStatus.Fail,
+        error_log: 'Organisation cannot be null or undefined',
+        output: {},
+      }
 
-    if (deletePermissions) await DBConnect.deleteUserOrgPermissions(userOrg)
+    // Remove single pair user-org if orgId and userId is defined
+    // Otherwise will remove ALL users from orgId (used when disable org)
+    if (userId)
+      console.log(`Removing user ${username ?? userId} from organisation ${orgName ?? orgId}...`)
+    else console.log(`Removing all users from organisation ${orgName ?? orgId}...`)
 
-    if (result.success) {
-      console.log(`User ${userOrg.userId} removed from organisation ${userOrg.orgId}`)
+    const removed: { pairs: UserOrg[]; success: boolean } = await DBConnect.removeUserOrg({
+      orgId,
+      userId,
+    })
+    if (deletePermissions) await DBConnect.deleteUserOrgPermissions({ orgId, userId })
+    if (removed.success) {
       return {
         status: ActionQueueStatus.Success,
         error_log: '',
-        output: {
-          userOrgId: result.id,
-          userId: result.user_id,
-          orgId: result.organisation_id,
-        },
+        output: removed.pairs,
```

Comment:
This has to be a named property, not an array. Cos the `output` from multiple actions is cumulatively built into the "cumulativeOutput" object by merging the output of each sequential action object. 

So this will break that.

Probably just do `output: {removedUsers: removed.pairs}`

Also -- can you have a think whether any of our current configs use the output of this action anywhere? Cos if so, those will need to be updated

Reply:
Oh, and remember to update the docs accordingly :)

Reply:
Oh, that makes a lot of sense! Thanks

No, I don't think this is used by any other action as an input